sqlglot/generator.py

```diff
@@ -142,6 +150,10 @@ def __init__(
         self.quote_end = quote_end or "'"
         self.identifier_start = identifier_start or '"'
         self.identifier_end = identifier_end or '"'
+        self.bit_string_start = bit_string_start
+        self.bit_string_end = bit_string_end
+        self.hex_string_start = hex_string_start
+        self.hex_string_end = hex_string_end
```

Comment:
The delimiter information is getting long. I wonder if it is cleaner to have default behavior here and overridden behavior as a lambda in dialect. For example, most languages (at least in their documentation) don't support bit and hex strings, so the default behavior would be to turn it into decimal; however, for the dialects that do support it, the dialect class will add a lambda that will be called instead, which maintains the hex base and adds the delimiters.

Reply:
sure, we can do that